command/meta.go

```diff
@@ -171,17 +172,59 @@ func (m *ApiMeta) Process(args []string) ([]string, error) {
 		return args, err
 	}
 
+	spinVersion := spinversion.String()
 	cfg := &gate.Configuration{
 		BasePath:      m.GateEndpoint(),
 		DefaultHeader: make(map[string]string),
-		UserAgent:     fmt.Sprintf("%s/%s", version.UserAgent, version.String()),
+		UserAgent:     fmt.Sprintf("%s/%s", spinversion.UserAgent, spinVersion),
 		HTTPClient:    client,
 	}
 	m.GateClient = gate.NewAPIClient(cfg)
 
+	err = m.gateVersionCheck()
+	if err != nil {
+		return nil, err
```

Comment:
What's the reasoning behind returning `nil, err` vs `args, err` elsewhere in the same function?